client/network/src/protocol/generic_proto/behaviour.rs

```diff
@@ -163,33 +187,37 @@ enum PeerState {
 	/// We are connected to this peer and the peerset has accepted it. The handler is in the
 	/// enabled state.
 	Enabled {
-		/// How we are connected to this peer.
-		connected_point: ConnectedPoint,
-		/// If true, we have a custom protocol open with this peer.
-		open: bool,
+		/// The connections that are currently open for custom protocol traffic.
+		open: SmallVec<[ConnectionId; crate::MAX_CONNECTIONS_PER_PEER]>,
 	},
 
-	/// We are connected to this peer, and we sent an incoming message to the peerset. The handler
-	/// is in initialization mode. We are waiting for the Accept or Reject from the peerset. There
-	/// is a corresponding entry in `incoming`.
-	Incoming {
-		/// How we are connected to this peer.
-		connected_point: ConnectedPoint,
-	},
+	/// We received an incoming connection from this peer and forwarded that
+	/// connection request to the peerset. The connection handlers are waiting
+	/// for initialisation, i.e. to be enabled or disabled based on whether
+	/// the peerset accepts or rejects the peer.
+	Incoming,
 }
 
 impl PeerState {
-	/// True if we have an open channel with that node.
+	/// True if there exists an established connection to tbe peer
+	/// that is open for custom protocol traffic.
 	fn is_open(&self) -> bool {
+		self.get_open().is_some()
+	}
+
+	/// Returns the connection ID of the first established connection
+	/// that is open for custom protocol traffic.
+	fn get_open(&self) -> Option<ConnectionId> {
 		match self {
-			PeerState::Poisoned => false,
-			PeerState::Banned { .. } => false,
-			PeerState::PendingRequest { .. } => false,
-			PeerState::Requested => false,
-			PeerState::Disabled { open, .. } => *open,
-			PeerState::DisabledPendingEnable { open, .. } => *open,
-			PeerState::Enabled { open, .. } => *open,
-			PeerState::Incoming { .. } => false,
+			PeerState::Disabled { open, .. } |
+			PeerState::DisabledPendingEnable { open, .. } |
+			PeerState::Enabled { open, .. } =>
+				if !open.is_empty() {
+					Some(open[0])
+				} else {
+					None
+				}
+			_ => None
```

Comment:
I prefer when we have all the variants explicitly written, as it makes it easier to see that we didn't forget something.

```suggestion
			PeerState::Poisoned => None,
			PeerState::Banned { .. } => None,
			PeerState::PendingRequest { .. } => None,
			PeerState::Requested => None,
			PeerState::Incoming { .. } => None,
```